sys/kern/sys_process.c

```diff
@@ -582,6 +583,19 @@ ptrace_sc_ret_to32(const struct ptrace_sc_ret *psr,
 }
 #endif /* COMPAT_FREEBSD32 */
 
+#if __has_feature(capabilities)
+static void
+ptrace_sc_ret_to64(const struct ptrace_sc_ret * __capability psr,
+    struct ptrace_sc_ret64 * __capability psr64)
+{
+
+	bzero((__cheri_fromcap void *)psr64, sizeof(*psr64));
```

Comment:
I think this will generate a CToPtr relative to $ddc which is not ideal. Would be nice to have a capability bzero.

Reply:
I agree that a bzero_c would be nice.  This pattern is used in several other places in this file currently so I was just reusing what was already there.

Reply:
The need for this reflects on the poor design of `kern_ptrace` where `addr` is overloaded and may be a user pointer or a kernel pointer.  I think this is fine for now, but we're overdue for an audit of `__cheri_fromcap` usage.

Reply:
In this case though it's more that we just don't have a bzero_c as the kernel capability is valid and points to an on-stack region.  Zeroing structures to avoid leaking kstack bits in padding will always be needed regardless of 'addr' being conflated.